skrub/_string_encoder.py

```diff
@@ -0,0 +1,195 @@
+        """
+        del y
+
+        if self.vectorizer == "tfidf":
+            self.vectorizer_ = TfidfVectorizer(
+                ngram_range=self.ngram_range, analyzer=self.analyzer
+            )
+        elif self.vectorizer == "hashing":
+            self.vectorizer_ = Pipeline(
+                [
+                    (
+                        "hashing",
+                        HashingVectorizer(
+                            ngram_range=self.ngram_range, analyzer=self.analyzer
+                        ),
+                    ),
+                    ("tfidf", TfidfTransformer()),
+                ]
+            )
+        else:
+            raise ValueError(f"Unknown vectorizer {self.vectorizer}. Options are 'tfidf' or 'hashing', got {self.vectorizer!r}")
+
+        X_out = self.vectorizer_.fit_transform(X)
+
+        if (min_shape := min(X_out.shape)) >= self.n_components:
+            self.tsvd_ = TruncatedSVD(n_components=self.n_components)
+            result = self.tsvd_.fit_transform(X_out)
+        else:
+            warnings.warn(
+                f"The matrix shape is {(X_out.shape)}, and its minimum is "
+                f"{min_shape}, which is too small to fit a truncated SVD with "
+                f"n_components={self.n_components}. "
+                "The embeddings will be truncated by keeping the first "
+                f"{self.n_components} dimensions instead. "
+            )
+            # self.n_components can be greater than the number
+            # of dimensions of result.
+            # Therefore, self.n_components_ below stores the resulting
+            # number of dimensions of result.
+            result = X_out[:, : self.n_components].toarray()
+
+        self._is_fitted = True
+        self.n_components_ = result.shape[1]
+
+        name = sbd.name(X)
+        if not name:
+            name = "tsvd"
+        self.all_outputs_ = [f"{name}_{idx}" for idx in range(self.n_components_)]
+
+        return self._transform(X, result)
+
+    def transform(self, X):
+        """Transform a column.
+
+        Parameters
+        ----------
+        X : Pandas or Polars series
+            The column to transform.
+
+        Returns
+        -------
+        result: Pandas or Polars dataframe with shape (len(X), tsvd_n_components)
+            The embedding representation of the input.
+        """
+
+        X_out = self.vectorizer_.transform(X)
+        if hasattr(self, "tsvd_"):
+            result = self.tsvd_.transform(X_out)
+        else:
+            result = X_out[:, : self.n_components].toarray()
+
+        return self._transform(X, result)
+
+    def _transform(self, X, result):
+        result = sbd.make_dataframe_like(X, dict(zip(self.all_outputs_, result.T)))
+        result = sbd.copy_index(X, result)
+
+        return result
+
+    def __sklearn_is_fitted__(self):
+        """
+        Check fitted status and return a Boolean value.
+        """
+        return check_is_fitted(self, "all_outputs_")
```

Comment:
it doesn't hurt to have it but this method is not needed: by default scikit-learn's `check_is_fitted` will look for attributes with name ending with an underscore, so it would return the same result whether we override the default behavior by defining __sklearn_is_fitted__ or not